lib/internal/bootstrap/loaders.js

```diff
@@ -193,6 +195,38 @@
     '\n});'
   ];
 
+  const { isProxy } = internalBinding('types');
+  const {
+    apply: ReflectApply,
+    has: ReflectHas,
+    get: ReflectGet,
+    set: ReflectSet,
+    defineProperty: ReflectDefineProperty,
+    deleteProperty: ReflectDeleteProperty,
+    getOwnPropertyDescriptor: ReflectGetOwnPropertyDescriptor,
+  } = Reflect;
+  const {
+    toString: ObjectToString,
+    hasOwnProperty: ObjectHasOwnProperty,
+  } = Object.prototype;
+  let isNative;
+  {
+    const { toString } = Function.prototype;
+    const re = toString.call(toString)
+      .replace(/[\\^$.*+?()[\]{}|]/g, '\\$&')
+      .replace(/toString|(function ).*?(?=\\\()/g, '$1.*?');
+    const nativeRegExp = new RegExp(`^${re}$`);
+    isNative = (fn) => {
+      if (typeof fn === 'function' &&
+          nativeRegExp.test(toString.call(fn))) {
+        const { name } = fn;
+        if (typeof name !== 'string' || !/^bound /.test(name))
```

Comment:
any particular reason? regex is still faster at this time

Reply:
👆 It's micro-opt territory here so it's up to personal preference is all.

Reply:
Interesting, I didn't realise that had changed in v8. Just paying attention to the getter hot path here.

Reply:
This very much does seem on the hot path to me? get -> wrap -> isNative is on every get in the proxy, so this path seems like it will hit every access to bound functions on exports?

Reply:
This `startsWith` check is done after the first few check so will only apply to the first access of bound functions and native functions. I don't think there's a need to get hung up on micro-optimizations at this point. If you find the need later then sure we can tweak various ways to detect xyz.